bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/src/main/java/com/google/cloud/bigtable/mapreduce/validation/BigtableSyncMapper.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.mapreduce.validation;
+
+import static com.google.cloud.bigtable.mapreduce.validation.BigtableSyncTableJob.SOURCE_BT_APP_PROFILE_CONF_KEY;
+import static com.google.cloud.bigtable.mapreduce.validation.BigtableSyncTableJob.SOURCE_BT_INSTANCE_CONF_KEY;
+import static com.google.cloud.bigtable.mapreduce.validation.BigtableSyncTableJob.SOURCE_BT_PROJECTID_CONF_KEY;
+import static com.google.cloud.bigtable.mapreduce.validation.BigtableSyncTableJob.TARGET_BT_APP_PROFILE_CONF_KEY;
+import static com.google.cloud.bigtable.mapreduce.validation.BigtableSyncTableJob.TARGET_BT_INSTANCE_CONF_KEY;
+import static com.google.cloud.bigtable.mapreduce.validation.BigtableSyncTableJob.TARGET_BT_PROJECTID_CONF_KEY;
+
+import com.google.cloud.bigtable.hbase.BigtableConfiguration;
+import com.google.cloud.bigtable.hbase.BigtableOptionsFactory;
+import java.io.IOException;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.client.ClusterConnection;
+import org.apache.hadoop.hbase.client.Connection;
+import org.apache.hadoop.hbase.client.Mutation;
+import org.apache.hadoop.hbase.client.Result;
+import org.apache.hadoop.hbase.io.ImmutableBytesWritable;
+import org.apache.hadoop.hbase.mapreduce.BigtableSyncTableAccessor;
+import org.apache.hadoop.hbase.mapreduce.SyncTable.SyncMapper;
+import org.apache.hadoop.mapreduce.Mapper;
+
+/** Bigtable SyncMapper */
+public class BigtableSyncMapper extends SyncMapper {
+
+  private static final Log LOG = LogFactory.getLog(BigtableSyncMapper.class);
+
+  @Override
+  public void setup(
+      Mapper<ImmutableBytesWritable, Result, ImmutableBytesWritable, Mutation>.Context context)
+      throws IOException {
+
+    // modify configuration to establish valid connection
+    Configuration conf = context.getConfiguration();
+    conf.unset(ClusterConnection.HBASE_CLIENT_CONNECTION_IMPL);
+
+    // since both source/target require establishing a connection, set the source as the target or
+    // target as source if not set and connecting with bigtable
```

Comment:
This needs a bit more explanation. If I understand correctly, this is necessary when this used bigtable is on one side of the sync. In that case super.setup() needs a fake config to initialize?

Reply:
good suggestion, updated and moved this logic this to its own method. 

your understanding is correct. we intend to use SyncMapper super.setup() to initialize the map task, but extend it to establish a connection with Bigtable if it's defined as a source or target. 

in parent [SyncTable.SyncMapper.setup(Context context)](https://github.com/apache/hbase/blob/master/hbase-mapreduce/src/main/java/org/apache/hadoop/hbase/mapreduce/SyncTable.java#L222-L252) a connection is established for creating an HBase connection to both source and target. if source zk or target zk are not set, which is the case when Bigtable is configured as source or target, the HBase connection defaults to localhost:2181 and will fail with establishing a connection.